grpc/server/src/chronicles.rs

```diff
@@ -385,14 +380,21 @@ fn read_chronicle_template(
     );
 
     // Process, the arguments of the action, adding them to the parameters of the chronicle and to the name of the action
-    for arg in action.parameters.clone() {
+    for (arg, arg_type) in action
+        .parameters
+        .clone()
+        .into_iter()
+        .zip(action.parameter_types.clone())
+    {
         let arg = Sym::from(arg.clone());
+        let arg_type = Sym::from(arg_type.clone());
         let tpe = context
             .model
             .get_symbol_table()
             .types
-            .id_of(&arg)
-            .ok_or_else(|| arg.invalid("Unknown atom"))?;
+            .id_of(&arg_type)
+            .ok_or_else(|| arg.invalid("Unknown argument"))?;
+        println!("base_name: {:?}", base_name);
         let arg = context.model.new_optional_sym_var(tpe, prez, c / VarType::Parameter); // arg.symbol
```

Comment:
@arbimo `at line 398` The action parameters cannot be converted into `SVar`.

Produces,
```
thread 'main' panicked at 'Variable with empty symbolic domain (note that we do not properly handle optionality in this case)', aries/model/src/model.rs:177:13
```

The file is at [model.rs](https://github.com/franklinselva/aries/blob/protobuf-interface/model/src/model.rs)

Reply:
Mmmh, there is a know bug (#27) that prevents variables from having an empty domain (even if the variable is not used).
Is there any object declared with this type `tpe` in the domain ?

Reply:
For the domain in #30, ti should not : `l_from` has the `Location` type which has 2 insances `l1` and `l2`. So maybe the problem is on the addition of these instances to the symbol table ?

Reply:
For the same problem, the instances and types are available in the symbol table. But the `ContiguousSymbols` for the type `Location` stays the same for both `first = 4` and `after_last = 4` which fails in the `ContiguousSymbols.bounds()` condition (the current type can never pass `first < after_last`).

```rust
pub struct ContiguousSymbols {
    first: usize,
    after_last: usize,
}
```

Reply:
Thank you. I am able to test some supported problems that have not yet been solved. I will try to create more simple examples for testing.